server/embed/config.go

```diff
@@ -772,8 +775,9 @@ func (cfg *Config) Validate() error {
 	}
 
 	switch cfg.AutoCompactionMode {
-	case "":
 	case CompactorModeRevision, CompactorModePeriodic:
+	case "":
+		return fmt.Errorf("undefined auto-compaction-mode")
```

Comment:
Minor comment, no need to use `fmt.Errorf` because there is no variadic parameter
```suggestion
		return errors.New("undefined auto-compaction-mode")
```